fix: make datagrid header sticky no matter how many rows

hippee-lee opened the pull request:
This change fixes a regression that breaks the sticky behavior of datagrid column headers.

Closes #4889

Signed-off-by: Matt Hippely <mhippely@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?
There was a regression introduced that broke the sticky column (header row) behavior in datagrid. This reverts that change. 

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently sticky headers will scroll out of view when the last visible row overflows the container element. This was happening because the `.datagrid` container had a display: flex on it. 
The code removed indicates that it was added to fix content stretching in Safari but I couldn't see what that issue was when it was removed. 

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #4889 

## What is the new behavior?
Datagrid headers stay in view when row content overflows no matter how much overflow there is. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

Jinnie reviewed (changes requested):
This simply unfixes the previous fix and will return the problem with no inner border in "Details" view when there are very few items in the list.
I researched more how we can fix the latter issue, without using flexbox on the parent and came with this solution, in which the flexbox is not needed:
<img width="570" alt="Screen Shot 2020-09-17 at 1 07 14 PM" src="https://user-images.githubusercontent.com/1798828/93456897-f5e7d800-f8e6-11ea-9ff7-898f7ac91faf.png">
which should be better, as we're resetting all the parents of the header individually, using only min-width recommendation, not forcing a strict width.
I checked if the datagrid looks OK with 0 and 1 items (because of the commented previous min-width) and it looks good.
It probably has to be tested for other regressions before commit.

Jinnie reviewed (commented):
This is an example of the missing border issue I'm talking about. Can be tested at http://localhost:4200/datagrid/detail

<img width="1011" alt="Screen Shot 2020-09-17 at 1 31 07 PM" src="https://user-images.githubusercontent.com/1798828/93459551-87a51480-f8ea-11ea-91b4-080a7b0a8faf.png">

hippee-lee commented:
Thanks @Jinnie - thats why I added you as a reviewer ... it wasn't clear to me why the flexbox had been added to the `.datagrid` container and I didn't see anything obviously wrong with the dev grids. I was hoping to get your insight to catch anything that wasn't obvious. Fwiw - the sticky header + sticky controls column is a little tricky and depends on the exact nested flex structure that we ended up with so that we can manage it with css (native browser) only. 

This highlights how important it is to get our visual regression testing back. When I refactored the datagrid for sticky header/sticky first (row controls) columns I relied heavily on our old Gemini testing to warn me of the side effects the refactor was having on the visual styles. 

Thanks again for double checking, I'll push changes up soon.

Jinnie commented on packages/angular/projects/clr-angular/src/data/datagrid/_datagrid.clarity.scss:
```diff
@@ -645,16 +643,15 @@
     flex-direction: column;
     flex: 1 1 auto;
     align-content: flex-start;
-    min-height: $clr_baselineRem_3; // Includes header and space for a single (non existing) row
```
It seems `min-height: 100%`, which I was suggesting, is not really needed here, that's good. 
We can probably return this safeguard to `$clr_baselineRem_3`?
I really don't know why it was included, and I don't like removing things like that, unless unavoidable. Or unexpected regressions may appear.

I am approving this, as it's not a blocker, just a precaution. If you decide (or know the exact origin of the safeguard) you can either return it or commit like this. Thanks for the fix and the fix of the fix.
